manual/manual/tutorials/coreexamples.etex

```diff
@@ -677,6 +677,36 @@ print_expr e; print_newline ();;
 print_expr (deriv e "x"); print_newline ();;
 \end{caml_example}
 
+There is a "printf" function in "Printf" module that allows you to make
+formatted output more concisely. It models the behavior of the "printf"
+function from the C standard library.
+The "printf" function takes a format string that describes the output values and
+their format, and arguments for output that correspond to the formatting
+directives in the format string.
+\begin{caml_eval}
+open Printf;;
+\end{caml_eval}
+\begin{caml_example}{toplevel}
+printf "%i + %i is an integer value, %F * %F is a float, \"%s\"\n"
+3 2 4.5 1. "this is a string";;
+\end{caml_example}
+The printf function in the OCaml language performs type checking of arguments.
+For example, if you pass it an argument of a type that does not correspond to
+the format specifier, the compiler will display an error message:
+\begin{caml_example}{toplevel}[error]
+printf "Float value: %F" 42;;
+\end{caml_example}
+Associated with this feature is the need to explicitly indicate that the
+string is a format string in order to pass it to the "printf" function.
+\begin{caml_example*}{toplevel}
+let str : ('a, 'b, 'c) format =
+    "%i is an integer value, %F is a float, \"%s\"\n";;
+\end{caml_example*}
+\begin{caml_example}{toplevel}
+printf str 3 4.5 "string value";;
+\end{caml_example}
+
```

Comment:
I think it would be interesting to get the previous pretty print example rewritten in terms of `Format` and `Printf`, showing how to create pretty printers and use them (even if in "real-life" I would probably rely on the `fmt` library)

Reply:
I concur. Having an example of a "%a" printer along the way would be nice too.